### PR TITLE
add r-bipartitemodularitymaximization recipe

### PR DESCRIPTION
**Pull request by ezherman**
Generated with conda_r_skeleton_helper

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by dbast — APPROVED** [review]

**Comment by jdblischak** [comment]
The package was successfully built on Windows ([source](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=683327&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=1499)):

```
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* MD5 sums
packaged installation of 'BipartiteModularityMaximization' as BipartiteModularityMaximization_1.23.120.1.zip
* DONE (BipartiteModularityMaximization)
```

The error occurs when conda-build checks for any missing system dependencies when ([source](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=683327&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=2939)):

```
  ERROR (r-bipartitemodularitymaximization,lib/R/library/BipartiteModularityMaximization/libs/x64/BipartiteModularityMaximization.dll): $RPATH/R.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
```

I wasn't able to find this exact error message before. However, the general strategy to get around this is to add it to `missing_dso_whitelist` so that conda-build knows it ok for this linked system library to be missing. As an example, this is done when [packaging CUDA on Windows](https://conda-forge.org/docs/maintainer/knowledge_base.html#nvcuda-dll-cannot-be-found-on-windows).

Searching GitHub, I could only find one recipe that adds `$RPATH/R.dll` to `missing_dso_whitelist`, [`r-xgboost`](https://github.com/AnacondaRecipes/xgboost-feedstock/blob/c4c37e90ba48b2d419e5604343aa10eb252ee03c/recipe/meta.yaml#L159):

```yaml
      missing_dso_whitelist:
        - $RPATH/R.dll                       # [win]
        - /usr/lib/libc++.1.dylib            # [osx]
        - lib/libgcc_s.so.1   # [linux]
        - lib/libstdc++.so.6  # [linux]
```

What I don't know is if it is ok for `$RPATH/R.dll` to be missing for this particular package. To be sure, I'd recommend copy-pasting the example code from the package [README](https://cran.r-project.org/web/packages/BipartiteModularityMaximization/readme/README.html) into a file called `run_tests.R`, which should automatically be run by conda-build during the test phase.

**Comment by ezherman** [comment]
Thank you so much @jdblischak! That seems to have done the trick. I really appreciate you taking the time to write instructions. 

@dbast, would you be able to approve these changes?

**Comment by jdblischak** [comment]
The tests didn't run. @ezherman I apologize, but I gave you the wrong file name. It needs to be `run_test.R`. I accidentally told you the plural version, `run_tests.R`, which wasn't recognized by conda-build

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-section

Could you please do a quick `git mv` to rename the test file?

**Comment by ezherman** [comment]
Thank you for pointing that out @jdblischak! I have renamed the file now.

**Review by jdblischak — APPROVED** [review]
The test file was detected and automatically detected for the osx-64 and win-64 builds. I don't know why this isn't recognized in the linux-64 build. But the main concern was the Windows build with the missing DSO, and the package clearly works

**Comment by ezherman** [comment]
Great, thank you for helping @jdblischak! @dbast, would you be able to merge these changes?

File: recipes/r-bipartitemodularitymaximization/run_test.R
Comment (mfansler) [inline recipes/r-bipartitemodularitymaximization/run_test.R:1]: This is simply a replication of what is already in the `testthat` tests. I'm going to remove this and have `testthat` run.

**Review by mfansler — CHANGES_REQUESTED** [review]

**Comment by mfansler** [comment]
@conda-forge-admin please rerender

**Comment by mfansler** [comment]
FYI, here is the reference on the `R.dll` issue: https://github.com/conda/conda-build/pull/4786

**Comment by mfansler** [comment]
I'll merge once everything passes 🤞

**Review by mfansler — APPROVED** [review]
All platforms now run the test and pass.

**Comment by mfansler** [comment]
@ezherman thanks for contributing this!

**Comment by ezherman** [comment]
Thank you for cleaning this up @mfansler!

**Pull request merged**
